proto/backend.proto

```diff
@@ -0,0 +1,159 @@
+syntax = "proto3";
+package pluginv2;
+
+//-----------------------------------------------
+// Common
+//-----------------------------------------------
+
+message PluginConfig {
+  int64 id = 1;
+  int64 orgId = 2;
+  string name = 3;
+  string type = 4;
+  string url = 5;
+  string jsonData = 6;
+  map<string,string> decryptedSecureJsonData = 7;
+}
+
+message TimeRange {
+  string fromRaw = 1;
+  string toRaw = 2;
+  int64 fromEpochMs = 3;
+  int64 toEpochMs = 4;
+}
+
+
+//-----------------------------------------------
+// QUERY
+//-----------------------------------------------
+
+message QueryRequest {
```

Comment:
I would love to see this model being used for both internal and external datasource implementations. If we ever feel that internal datasources needs something extra we can use the headers field to pass more info.

Reply:
yes exactly!